numpy/core/overrides.py

```diff
@@ -163,13 +162,13 @@ def decorator(implementation):
         # more interpettable name. Otherwise, the original function does not
         # show up at all in many cases, e.g., if it's written in C or if the
         # dispatcher gets an invalid keyword argument.
-        source = textwrap.dedent("""
-        @functools.wraps(implementation)
-        def {name}(*args, **kwargs):
-            relevant_args = dispatcher(*args, **kwargs)
-            return implement_array_function(
-                implementation, {name}, relevant_args, args, kwargs)
-        """).format(name=implementation.__name__)
+        source = """
+@functools.wraps(implementation)
+def {name}(*args, **kwargs):
+    relevant_args = dispatcher(*args, **kwargs)
+    return implement_array_function(
+        implementation, {name}, relevant_args, args, kwargs)
+""".format(name=implementation.__name__)
```

Comment:
I suspect this is where all the cost is - we end up dedenting the same string for every single numpy function, which is obviously expensive. Dedenting it just once wouldn't be nearly as bad, although at that point chances are the string definition lives outside the function, at which point there's nothing to even dedent.

In the interest of preserving the visual indent, I might be inclined to move this string definition to a global `_wrapped_func_source` variable in the file, which still avoids paying for textwrap

Reply:
> In the interest of preserving the visual indent, I might be inclined to move this string definition to a global _wrapped_func_source variable in the file, which still avoids paying for textwrap

Is this an opinion, or a requirement to get your approval on this PR?

Reply:
But I do suspect you are right in the source of the problem. I just don't personally see any issue with deindenting strings manually.

Reply:
> Is this an opinion

It's an opinion that probably gates my immediate approval until another maintainer weighs ins.

Reply:
I'm over it, lets hunt more milliseconds, and work to deprecate the whole testing module. Much more interesting ;)

Reply:
I have a problem with dedenting strings manually, they are ugly, and ugly is unpleasant to maintain.

Reply:
very fair.